src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -85,8 +85,66 @@ export const survivalClinicalDataNullValueSet = new Set([
     'na',
 ]);
 
+// Concept of Number at risk:
+// In order to calculate survival probability using the Kaplan-Meier product limit method
+// We need to know how many individuals were still accounted for in the study that had not yet experienced the event of interest.
+// Therefore, the number at risk at any specific time point will be equal to the total number of subjects remaining in the study,
+// including any individuals that experience the event of interest or individuals that are censored at this time point.
+export function getNumPatientsAtRisk(
+    // In order to calculate correct result
+    // Patient survivals need to be sorted by months and status
+    // First sort by month in asc order (smaller number to the front)
+    // Then sort by status in desc order (which means if status are '1's, then go to the front, '0's are right after '1's in the same time stamp)
+    sortedPatientSurvivals: Pick<PatientSurvival, 'entryMonths' | 'months'>[]
+): number[] {
+    // returns an array in the same order as patientSurvivals
+    const atRisk: number[] = [];
+
+    const entryTimes = _.sortBy(sortedPatientSurvivals.map(s => s.entryMonths));
+
+    // When we see one entry time is not zero, we consider the data is left truncated
+    if (_.some(entryTimes, entryTime => entryTime !== 0)) {
+        // Calculate number of patients at risk for data with entry time (left truncation)
+        let entryIndex = 0;
+        let atRiskAccumulator = 0;
+        _.reduce(
+            sortedPatientSurvivals,
+            (atRisk, patientSurvival) => {
+                const exitTime = patientSurvival.months;
+                // This step is to count the total number of subjects remaining in the study at this specific exit time
+                // fast forward through entries until we get past exit time
+                while (
```

Comment:
see how the use of the while and multiple incrementors confuses things.  i think you are just trying to get the count of entryTimes members which are less than exitTime and add that count to the accuulator  
```
atRiskAccumulator += entryTimes.filter(t=>t>extitTime).length
```

Reply:
This is true, thanks for the suggestion, I modified this based on the suggestion. One thing to consider is that using index can ignore repeating search (O(n) time complexity), and if we filter every time, it will be O(n^2). So we can keep this in mind, and if we experiencing performance issue later, I can change to use index approach if needed.